cdap-support-bundle/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java

```diff
@@ -111,4 +137,108 @@ public void createSupportBundle(HttpRequest request, HttpResponder responder,
       responder.sendString(HttpResponseStatus.OK, prevInProgressUUID);
     }
   }
+
+  /**
+   * Get the list of bundle status from status.json.
+   *
+   * @throws NotFoundException is thrown if status.json is not found
+   */
+  @GET
+  @Path("/support/bundles")
+  public void listSupportBundles(HttpRequest request, HttpResponder responder) throws Exception {
+    // Ensure the user is authorized to list support bundle statuses
+    contextAccessEnforcer.enforceOnParent(EntityType.SUPPORT_BUNDLE, InstanceId.SELF, StandardPermission.LIST);
+    File baseDirectory = new File(cConf.get(Constants.SupportBundle.LOCAL_DATA_DIR));
+    if (!baseDirectory.exists()) {
+      LOG.debug("No content in Support Bundle.");
+      return;
+    }
+    List<SupportBundleOperationStatus> bundleOperationStatusList = new ArrayList<>();
+    DirUtils.listFiles(baseDirectory)
+      .stream()
+      .filter(file -> !file.isHidden() && file.isDirectory())
+      .forEach(uuidFile -> {
+        SupportBundleOperationStatus bundleOperationStatus = null;
+        try {
+          bundleOperationStatus = bundleGenerator.getBundle(uuidFile.getName());
+        } catch (IOException e) {
+          LOG.debug("Can not find status json: {}", uuidFile.getName());
+        }
+        bundleOperationStatusList.add(bundleOperationStatus);
```

Comment:
So essentially you are adding a null to the list. Is that what you wanted??

Reply:
yeah, either that or i can initiate it with an invalid status